### PR TITLE
Only init gatewayapi client when enabled

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @ramaraochavali

**Review by howardjohn — COMMENTED** [review]
Is this fixing any issues? AFAIK creating the client and informerfactory is ~free, and calling .Run is also free unless an informer is registered

**Comment by hzxuzhonghu** [comment]
@howardjohn As you see the informer is started

**Comment by howardjohn** [comment]
.Run starts the informer *factory* not the informer. If there are no informer registered it does nothing.

If it did start the informer then you would get tons of errors if you didn't have the CRD installed

File: pkg/kube/client.go [hunk 1]
```diff
@@ -459,7 +464,9 @@ func (c *client) RunAndWait(stop <-chan struct{}) {
 	c.dynamicInformer.Start(stop)
 	c.metadataInformer.Start(stop)
 	c.istioInformer.Start(stop)
-	c.gatewayapiInformer.Start(stop)
```
Comment (hzxuzhonghu) [inline pkg/kube/client.go:462]: @howardjohn What does this mean if it does not start gatewayapiInformer

**Comment by hzxuzhonghu** [comment]
@howardjohn you are right, this is just an informer factory

**Pull request closed without merging**